modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesViewModel.kt

```diff
@@ -139,7 +142,14 @@ class StoriesViewModel @Inject constructor(
                 EOY_STORY_SAVE_FOLDER_NAME,
                 EOY_STORY_SAVE_FILE_NAME
             )
-            savedFile?.let { showShareForFile.invoke(it) }
+
+            val currentState = (state.value as State.Loaded)
+            mutableState.value = currentState.copy(preparingShareText = true)
+
+            val shareTextData = shareableTextProvider.getShareableDataForStory(story)
+            mutableState.value = currentState.copy(preparingShareText = false)
```

Comment:
This is because `StoriesViewModel` initialized inside the `StoriesPage` dialog composable is not scoped to it but to the activity/ fragment that contains it. As a workaround, I manually call viewmodel's `clear()` and `start()` which doesn't seem to be working well. 

This [SO answer](https://stackoverflow.com/a/67321848/193545) helped me understand it:
> Compose does not offer any mechanism to scope ViewModels to an individual @Composable - any ViewModels you create outside of a NavHost's destination is scoped to the activity/fragment that contains your ComposeView/where you call setContent and, thus, lives for the entire lifetime of your Compose hierarchy - that's why you always get the same instance back.

Feature request: https://issuetracker.google.com/issues/165642391

I'll open another one with a fix or temporary workaround.

Reply:
Do you think that disabling the UI as soon as the Share button is touched might help?

Reply:
I switched to `DialogFragment` with composable content in https://github.com/Automattic/pocket-casts-android/pull/581. I'll shortly make it ready for review and we can test it again if it fixes the issue.